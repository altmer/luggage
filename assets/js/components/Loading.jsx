import React from 'react';
import Loader from 'halogen/PulseLoader';

export const Loading = () => (
  <Loader color="#0275d8" size="12px" margin="4px" />
);

export default Loading;
