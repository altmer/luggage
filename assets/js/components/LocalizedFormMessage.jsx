import React from 'react';
import { I18n } from 'react-i18nify';
import Form from 'react-formal';

const extract = (error) => {
  const key = error.message || error;
  return I18n.t(`errors.${key}`);
};

export const LocalizedFormMessage = props => (
  <Form.Message {...props} extract={extract} />
);

export default LocalizedFormMessage;
