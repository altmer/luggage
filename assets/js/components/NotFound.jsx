import React from 'react';
import { Translate } from 'react-i18nify';

export const NotFound = () => (
  <div className="container">
    <div className="row">
      <div className="col-12">
        <h3><Translate value="not_found.header" /></h3>
        <p><Translate value="not_found.description" /></p>
      </div>
    </div>
  </div>
);

export default NotFound;
