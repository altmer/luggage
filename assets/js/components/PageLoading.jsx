import React from 'react';
import Loader from 'halogen/RingLoader';

export const PageLoading = () => (
  <div className="container">
    <div className="row">
      <div className="col-12">
        <div className="page-loading">
          <Loader color="#0275d8" size="128px" />
        </div>
      </div>
    </div>
  </div>
);

export default PageLoading;
