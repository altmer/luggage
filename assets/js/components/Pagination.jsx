import React from 'react';
import PropTypes from 'prop-types';

import { Pagination as BootstrapPagination, PaginationItem, PaginationLink } from 'reactstrap';

export const Pagination = props => (
  <BootstrapPagination>
    {(props.page > 1) && <PaginationItem>
      <PaginationLink
        tag="button"
        previous
        onClick={() => props.onPaginate(props.page - 1)}
      />
    </PaginationItem>}
    {(props.page > 1) && <PaginationItem>
      <PaginationLink
        tag="button"
        onClick={() => props.onPaginate(props.page - 1)}
      >
        {props.page - 1}
      </PaginationLink>
    </PaginationItem>}
    <PaginationItem active>
      <PaginationLink tag="button">
        {props.page}
      </PaginationLink>
    </PaginationItem>
    {(props.page < props.totalPages) && <PaginationItem>
      <PaginationLink
        tag="button"
        onClick={() => props.onPaginate(props.page + 1)}
      >
        {props.page + 1}
      </PaginationLink>
    </PaginationItem>}
    {(props.page < props.totalPages) && <PaginationItem>
      <PaginationLink
        tag="button"
        next
        onClick={() => props.onPaginate(props.page + 1)}
      />
    </PaginationItem>}
  </BootstrapPagination>
);

Pagination.propTypes = {
  page: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  onPaginate: PropTypes.func.isRequired,
};

export default Pagination;
