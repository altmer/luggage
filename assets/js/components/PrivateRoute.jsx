import React from 'react';
import PropTypes from 'prop-types';

import { Route, Redirect } from 'react-router-dom';

const isAuthenticated = () => (!!localStorage.getItem('phoenixAuthToken'));

const PrivateRoute = ({ component, ...rest }) => (
  <Route
    {...rest}
    render={props => (
      isAuthenticated() ? (
        React.createElement(component, props)
      ) : (
        <Redirect to="/sign_in" />
      )
    )}
  />
);

PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired,
};

export default PrivateRoute;
