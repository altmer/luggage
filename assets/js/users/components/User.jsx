import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';

import Avatar from './Avatar';

export const User = props => (
  <div className="user-badge">
    <div className="user-badge-avatar">
      <Link to={`/users/${props.id}/show`}>
        <Avatar color={props.color} initials={props.initials} avatar={props.avatar} />
      </Link>
    </div>
    <div className="user-badge-content">
      <Link to={`/users/${props.id}/show`} className="user-badge-name">
        {props.name}
      </Link>
      {props.children}
    </div>
  </div>
);

User.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  initials: PropTypes.string.isRequired,
  avatar: PropTypes.shape({
    thumb: PropTypes.string,
  }),
  children: PropTypes.node,
};

User.defaultProps = {
  avatar: {
    thumb: null,
  },
  children: null,
};

export default User;
