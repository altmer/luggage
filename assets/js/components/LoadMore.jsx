import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-i18nify';

import Loading from './Loading';

export const LoadMore = props => (
  <div className="row">
    <div className="col-12">
      {props.show && !props.fetching &&
      <a
        href="#0"
        onClick={props.onClick}
        className="load-more-link"
      >
        <Translate value="common.loadMore" />
      </a>
      }
      {props.show && props.fetching &&
        <Loading />
      }
    </div>
  </div>
);

LoadMore.propTypes = {
  fetching: PropTypes.bool.isRequired,
  show: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default LoadMore;
