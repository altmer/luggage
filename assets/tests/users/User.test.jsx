import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import UserFactory from '../factories/user';
import { User } from '../../js/users/components/User';

const user = UserFactory.build();

test('it renders user information', () => {
  const subject = renderer.create(<MemoryRouter>
    <User {...user} />
  </MemoryRouter>);
  expect(subject.toJSON()).toMatchSnapshot();
});
