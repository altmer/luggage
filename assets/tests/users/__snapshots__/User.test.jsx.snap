// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders user information 1`] = `
<div
  className="user-badge"
>
  <div
    className="user-badge-avatar"
  >
    <a
      href="/users/1/show"
      onClick={[Function]}
    >
      <span>
        <div
          className="user-avatar user-avatar-placeholder 2"
        >
          JD
        </div>
      </span>
    </a>
  </div>
  <div
    className="user-badge-content"
  >
    <a
      className="user-badge-name"
      href="/users/1/show"
      onClick={[Function]}
    >
      John Doe
    </a>
  </div>
</div>
`;
